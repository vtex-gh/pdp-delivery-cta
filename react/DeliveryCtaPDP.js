import DeliveryCtaPDP from './components/DeliveryCtaPDP/index'

export default DeliveryCtaPDP
