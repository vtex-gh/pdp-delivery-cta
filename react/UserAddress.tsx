import React, { useEffect } from 'react'
import { useQuery } from 'react-apollo'
import { useCssHandles } from 'vtex.css-handles'
import type { CssHandlesTypes } from 'vtex.css-handles'
import ADDRESS_QUERY from 'vtex.store-resources/QueryAddress'

import Container from './Container'
import AddressInfo, {
  CSS_HANDLES as AddressInfoCssHandles,
} from './components/UserAddress/AddressInfo'
import { UserAddressCssHandlesProvider } from './components/UserAddress/UserAddressCssHandles'

export const USER_ADDRESS_CSS_HANDLES = [
  'userAddressContainer',
  ...AddressInfoCssHandles,
] as const

type Props = {
  /** Define if address should be displayed in a single line or in blocks */
  variation: 'inline' | 'bar'
  /** Define if street should be displayed */
  showStreet?: boolean
  /** Define if city and state should be displayed */
  showCityAndState?: boolean
  /** Define if postal code should be displayed */
  showPostalCode?: boolean
  /** Define if prefix should be displayed */
  showPrefix?: boolean
  /** Define if component should render if address is empty */
  showIfEmpty?: boolean
  /** Used to override default CSS handles */
  classes?: CssHandlesTypes.CustomClasses<typeof USER_ADDRESS_CSS_HANDLES>
}

function UserAddress({
  variation,
  showStreet = true,
  showCityAndState = false,
  showPostalCode = false,
  showPrefix = true,
  showIfEmpty = false,
  classes,
}: Props) {
  const { handles, withModifiers } = useCssHandles(USER_ADDRESS_CSS_HANDLES, {
    classes,
  })

  // todo: type the data returned
  const { data: orderFormData, refetch } = useQuery(ADDRESS_QUERY, {
    ssr: false,
  })

  useEffect(() => {
    const handleLocationUpdated = () => refetch()

    window.addEventListener('locationUpdated', handleLocationUpdated)

    return () => {
      window.removeEventListener('locationUpdated', handleLocationUpdated)
    }
  }, [refetch])

  const orderForm = orderFormData?.orderForm

  if (!orderForm || (!orderForm?.shippingData?.address && !showIfEmpty)) {
    return null
  }

  const isInline = variation === 'inline'
  const isInverted = !isInline

  const AddressInfoWithHandles = () => {
    return (
      <UserAddressCssHandlesProvider
        handles={handles}
        withModifiers={withModifiers}
      >
        <AddressInfo
          inverted={isInverted}
          inline={isInline}
          orderForm={orderForm}
          showStreet={showStreet}
          showCityAndState={showCityAndState}
          showPostalCode={showPostalCode}
          showPrefix={showPrefix}
          showIfEmpty={showIfEmpty}
        />
      </UserAddressCssHandlesProvider>
    )
  }

  if (isInline) {
    return (
      <div
        className={`${handles.userAddressContainer} ph5`}
        style={{
          maxWidth: '30rem',
        }}
      >
        <AddressInfoWithHandles />
      </div>
    )
  }

  return (
    <div
      className={`${handles.userAddressContainer} bg-base--inverted c-on-base--inverted flex ph5 pointer pv3 ml3 mr3`}
    >
      <Container className="flex justify-center w-100 left-0">
        <div className="w-100 mw9 flex">
          <AddressInfoWithHandles />
        </div>
      </Container>
    </div>
  )
}

export default UserAddress
