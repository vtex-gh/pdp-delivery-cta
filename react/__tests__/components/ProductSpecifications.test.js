import React from 'react'
import { render } from '@vtex/test-tools/react'

import ProductSpecifications from '../../ProductSpecifications'

describe('<ProductSpecifications /> component', () => {
  const renderComponent = customProps => {
    const props = {
      ...customProps,
    }

    const comp = <ProductSpecifications {...props} />

    return render(comp)
  }

  it('should be rendered', () => {
    const { asFragment } = renderComponent()

    expect(asFragment()).toBeTruthy()
  })

  it('should match snapshot with table view and no specifications', () => {
    const { asFragment } = renderComponent()

    expect(asFragment()).toMatchSnapshot()
  })

  it('should match snapshot with table view and with specification', () => {
    const { asFragment } = renderComponent({
      specifications: [{ name: 'test', values: ['value'] }],
    })

    expect(asFragment()).toMatchSnapshot()
  })

  it('should match snapshot with tabs view', () => {
    const { asFragment } = renderComponent({
      specifications: [{ name: 'test', values: ['value'] }],
      tabsMode: true,
    })

    expect(asFragment()).toMatchSnapshot()
  })
})
