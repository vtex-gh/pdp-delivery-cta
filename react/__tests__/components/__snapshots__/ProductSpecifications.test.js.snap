// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductSpecifications /> component should match snapshot with table view and no specifications 1`] = `<DocumentFragment />`;

exports[`<ProductSpecifications /> component should match snapshot with table view and with specification 1`] = `
<DocumentFragment>
  <div
    class="specificationsTableContainer mt9 mt0-l pl8-l"
  >
    <h2
      class="specificationsTitle t-heading-5 mb5 mt0"
    >
      <div
        style="display: contents;"
      >
        Technical Specifications
      </div>
    </h2>
    <div
      class="container relative"
      style="transition: 600ms ease-in-out; height: auto; overflow: hidden; display: block;"
    >
      <div
        class="content h-auto"
      >
        <table
          class="specificationsTable w-100 bg-base border-collapse"
        >
          <thead
            class="specificationsTableHead"
          >
            <tr
              class="specificationsTableRow"
            >
              <th
                class="specificationsTablePropertyHeading w-50 b--muted-4 bb bt c-muted-2 t-body tl pa5"
              >
                Property
              </th>
              <th
                class="specificationsTableSpecificationHeading w-50 b--muted-4 bb bt c-muted-2 t-body tl pa5"
              >
                Specification
              </th>
            </tr>
          </thead>
          <tbody
            class="specificationsTableBody"
          >
            <tr
              class="specificationsTableRow"
            >
              <td
                class="specificationItemProperty w-50 b--muted-4 bb pa5"
                data-specification="test"
              >
                <div
                  style="display: contents;"
                >
                  test
                </div>
              </td>
              <td
                class="specificationItemSpecifications w-50 b--muted-4 bb pa5"
                data-specification="value"
              >
                <div
                  style="display: contents;"
                >
                  value
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="pointerEventsNone dn absolute bottom-0 w-100 h-100 flex-column justify-end"
      >
        <div
          class="fadeBottom w-100 h-50"
          style="transition: 400ms ease-in-out;"
        />
        <div
          class="pointerEventsAuto bg-base tc w-100"
        >
          <button
            class="showMoreButton c-action-primary t-action pointer ma5 bn outline-0"
          >
            Show more
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<ProductSpecifications /> component should match snapshot with tabs view 1`] = `
<DocumentFragment>
  <div
    class="specificationsTabsContainer pt8"
  >
    <h2
      class="specificationsTitle t-heading-5 mb5 mt0"
    >
      <div
        style="display: contents;"
      >
        Technical Specifications
      </div>
    </h2>
    <div
      class="tabs-mock"
    >
       
      <div
        class="tab-mock"
      >
         
        <div
          class="specificationsTab pb8 c-muted-1 pv6"
        >
          <div
            class="container relative"
            style="transition: 600ms ease-in-out; height: auto; overflow: hidden; display: block;"
          >
            <div
              class="content h-auto"
            >
              <div
                style="display: contents;"
              >
                value
              </div>
            </div>
            <div
              class="pointerEventsNone dn absolute bottom-0 w-100 h-100 flex-column justify-end"
            >
              <div
                class="fadeBottom w-100 h-50"
                style="transition: 400ms ease-in-out;"
              />
              <div
                class="pointerEventsAuto bg-base tc w-100"
              >
                <button
                  class="showMoreButton c-action-primary t-action pointer ma5 bn outline-0"
                >
                  Show more
                </button>
              </div>
            </div>
          </div>
        </div>
         
      </div>
       
    </div>
  </div>
</DocumentFragment>
`;
